ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/ktx/ColorScheme.kt

```diff
@@ -0,0 +1,11 @@
+@file:JvmSynthetic
```

Comment:
Hmm this is going to be internal right? I think we might not need this? As in, I would probably only add it on files that will contain public APIs but we don't want to be accessible from Java. Wdyt?

Reply:
Anything internal is public in Java, because Java doesn't have the internal keyword. So that's why I've been adding it to internal APIs as well. The `@file:` scope is [supposedly avoiding the creation of a facade class](https://youtrack.jetbrains.com/issue/KT-41884) (`ColorSchemeKt`) that would otherwise be generated.

Reply:
Internal symbols go through name mangling in Java, so something like this:
```kotlin
internal val default: Padding
```

becomes: 

```java
public Padding getDefault$purchases_defaultsRelease()
```

I think we should avoid that as much as possible, even if we're already doing this in some places. (The name can be fixed with `@JvmName`, but why not hide it completely at that point.)

>this reduces readability

I agree, but that's our problem. I'm afraid that we make it the developer's problem if we leak internal symbols to Java, which is aggravated by the name mangling. 

>why isn't the kotlin compiler doing this automatically upon finding any internal class/property? 

I'm assuming because you might want to access internal symbols from Java code in the same module. That would be impossible if everything internal is automatically hidden from Java.

For reference, the documentation on this behavior is [here](https://kotlinlang.org/docs/java-to-kotlin-interop.html#visibility).

Reply:
Yeah, I've added this topic to our weekly so we can discuss it... I'm not sure it's worth the extra overhead to try to hide these APIs from Java...